www/webkit2-gtk/Portfile

```diff
@@ -215,42 +249,6 @@ if {${os.platform} eq "darwin" && ${os.major} <= 13} {
     }
 }
 
-variant quartz conflicts x11 gtk2 {
-    require_active_variants port:gtk3 quartz
-
-    configure.args-append \
-        -DENABLE_QUARTZ_TARGET=ON \
-        -DENABLE_X11_TARGET=OFF \
-        -DENABLE_OPENGL=OFF \
-        -DENABLE_GRAPHICS_CONTEXT_3D=OFF
-}
-
-variant x11 conflicts quartz {
-    require_active_variants port:gtk3 x11
-
-    configure.args-append \
-        -DENABLE_QUARTZ_TARGET=OFF \
-        -DENABLE_X11_TARGET=ON \
-        -DENABLE_OPENGL=ON
-
-    depends_lib-append \
-        port:mesa \
-        port:xorg-libXt
-}
-
-variant minibrowser description {Build and install MiniBrowser (for testing)} {
-    configure.args-delete   -DENABLE_MINIBROWSER=OFF
-    configure.args-append   -DENABLE_MINIBROWSER=ON
-}
-
-variant gtk2 conflicts quartz description {Enable support for GTK+2 plugins} {
-    require_active_variants port:gtk2 x11
-
-    depends_lib-append      port:gtk2
-    configure.args-delete   -DENABLE_PLUGIN_PROCESS_GTK2=OFF
-    configure.args-append   -DENABLE_PLUGIN_PROCESS_GTK2=ON
-}
-
 livecheck.type      regex
 livecheck.url       http://webkitgtk.org/releases/
-livecheck.regex     "webkitgtk-(\\d+\\.\\d*\[02468\](?:\\.\\d+)*)"
+livecheck.regex     "webkitgtk-(\\d+(?:\\.\\d+)+)"
```

Comment:
Plural minor versions are stable releases, we should go for 2.22.6 instead.

Reply:
To be noted that this port is updated irregularly, due to it's somewhat difficult patchset, and the fact that it takes a surprising number of hours to build out on all systems for testing.

We never had a 2.22.x version in the devel series, this 2.21.91 version was the last we had, dating from the last time I had any serious time off work. I tried 2.21.91 but it would not (immediately) build with the patches here.

Instead of working on the the 2.22.x series, I jumped the webkit2-gtk-devel port to the latest 2.23.3, and that was going to be coming up in a new PR next.

If we wanted the 2.22.6 stable release for our webkit2-gtk, we'd have to do that as a fresh new untested version for us -- it would have patches likely somewhere between this version and 2.23.3. It's doable, but I was personally planning on skipping the 2.22.x series.

Things would be easier if I could upstream some of the patches. Jeremy indicated that might not be welcomed at one point, but that was a while ago now.

Reply:
changed my mind. Perhaps the better idea is to try for a 2.22.6 release as webkit2-gtk after all, and at least try to keep the pattern going.

Reply:
> To be noted that this port is updated irregularly, due to it's somewhat difficult patchset, and the fact that it takes a surprising number of hours to build out on all systems for testing.
> 
> We never had a 2.22.x version in the devel series, this 2.21.91 version was the last we had, dating from the last time I had any serious time off work. I tried 2.21.92 but it would not (immediately) build with the patches here.
> 
> Instead of working on the the 2.22.x series, I jumped the webkit2-gtk-devel port to the latest 2.23.3, and that was going to be coming up in a new PR next.

Yes, 2.23.x is the current unstable release, so the -devel port should track that.

I believe we should not provide any unstable (including alpha, rc, etc.) releases except in `*-devel` ports, unless the last stable release lags behind a lot and misses a lot of important features or bugfixes, or that no stable release has been announced yet.

> changed my mind. Perhaps the better idea is to try for a 2.22.6 release as webkit2-gtk after all, and at least try to keep the pattern going. Thanks for the lookover.

Thank you for maintaining this port, 2.22.6 would bring many security fixes (see [WSA-2019-0001](https://webkitgtk.org/security/WSA-2019-0001.html), WSA-2018-0009, and WSA-2018-0008) while 2.21.91 (i.e. 2.22.0-rc2) is still vulnerable. Have a maintainer working on these security issues is greatly appreciated.